### PR TITLE
fix(failure-rate-metrics-in-peformance): Fix wrong manual calculation

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Fix wrong failure rate calculation
- [x] Removes temp aggregation prop

### CODE REVIEW

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx [hunk 1]
```diff
@@ -69,59 +69,94 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
...
+              const totalSerieBucket = totalPerBucket?.[intervalIndex];
+              const totalFailureSerieBucket = totalFailurePerBucket?.[intervalIndex];
+
+              return {
+                name: moment(intervalValue).valueOf(),
+                value:
+                  defined(totalFailureSerieBucket) &&
+                  defined(totalSerieBucket) &&
+                  totalSerieBucket > 0 &&
+                  totalFailureSerieBucket > 0
```
Comment (matejminar) [inline static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:96]: We need to remove this line, the failure count can be 0.

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx [hunk 2]
```diff
@@ -69,59 +69,94 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
...
+
   const dataMean = data.map(serie => {
     let meanData: undefined | number = undefined;
 
     if (
       defined(seriesTotal) &&
-      defined(serie.totals) &&
-      serie.totals > 0 &&
+      defined(seriesTotalFailure) &&
+      seriesTotalFailure > 0 &&
```
Comment (matejminar) [inline static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:141]: Same here, we can get rid of this line, failure count can be 0.

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx [hunk 3]
```diff
@@ -69,59 +69,94 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
...
+            const serieBucket = group.series[metricsField][intervalIndex];
+            const totalSerieBucket = totalPerBucket?.[intervalIndex];
+            return {
+              name: moment(intervalValue).valueOf(),
+              value:
+                defined(totalSerieBucket) &&
+                defined(serieBucket) &&
+                serieBucket > 0 &&
+                totalSerieBucket > 0
+                  ? serieBucket / totalSerieBucket
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:114]: good catch!

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:96]: I have these conditions to avoid `NaN` and `Infinity`

Comment (matejminar) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:96]: We can still check if totalFailureSerieBucket is a number, but it does not need to be greater than 0.
Consider this case: We have 20 total transactions, 0 failed. We want the failure rate to be 0/20 -> 0, not 20.

Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx]: but we can monitor it

Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx]: and if both are 0? example `0/0`, then we would get `NaN`

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx [hunk 4]
```diff
@@ -54,108 +54,147 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
 
   const isFailureRateWidget = chartSetting === PerformanceWidgetSetting.FAILURE_RATE_AREA;
 
-  const groups = isFailureRateWidget
-    ? response.groups.filter(
-        group => !TRANSACTION_SUCCESS_STATUS.includes(group.by['transaction.status'])
+  if (isFailureRateWidget) {
+    const groups = response.groups.filter(
```
Comment (matejminar) [inline static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx]: How about we rename this to failedGroups for clarity?

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx [hunk 5]
```diff
@@ -54,108 +54,147 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
...
-      )
-    : responsePrevious?.groups;
+      return {
+        mean: meanData,
+        outputType: aggregateOutputType(serie.seriesName),
+        label: axisLabelFormatter(meanData, serie.seriesName),
+      };
+    });
+
+    const previousGroups = responsePrevious?.groups.filter(
```
Comment (matejminar) [inline static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:114]: Also maybe rename to previousFailedGroups?

**Review by matejminar — APPROVED** [review]
I checked the data and now it looks right 👍

**Pull request merged**